stl/inc/filesystem

```diff
@@ -1377,6 +1377,36 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Handle the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {

```

Comment:
This assumes `_Right_size != 0`, which I don't think we've verified.
```suggestion
            if (_Right_Size != 0 && !_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
```

Reply:
Agreed, great catch. (`basic_string` null termination probably saves us, but it's squirrely for sure.) If we were positively testing `_Has_drive_letter_prefix` then we'd get to assume `_Last - _First >= 2`, but the test here is negated, so we don't know anything about `_Right`'s size before saying `*_Right_first` at the end.

I'll apply your suggested change (with a typo fix).